lms/djangoapps/courseware/views/index.py

```diff
@@ -182,6 +181,22 @@ def _setup_masquerade_for_effective_user(self):
         # Set the user in the request to the effective user.
         self.request.user = self.effective_user
 
+    def _redirect_to_learning_mfe(self, request):
+        """
+        Redirect to the new courseware micro frontend,
+        unless this is a time limited exam.
+
+        TODO: remove this once exams work in the new MFE.
+        """
+        if (not self.section.is_time_limited) \
+                and should_redirect_to_courseware_microfrontend(self.course_key) \
+                and not request.user.is_staff:
```

Comment:
1. Does this properly exempt course staff?
2. This is a kind of complex boolean statement. It might be easier to read you break it out into multiple clauses that return early, e.g.:

```python
# Global staff are never redirected...
if request.user.is_staff:
    return

# Don't redirect for time limited exams (until we add support for this later)
if self.section.is_time_limited:
   return

# etc...
```

Reply:
(item 2 here is a suggestion, not a requirement to merge)